client/interface.go

```diff
@@ -0,0 +1,33 @@
+package client
+
+import (
+	"encoding"
+
+	"github.com/libp2p/go-libp2p-core/event"
+)
+
+// Client represents the drand Client interface.
+type Client interface {
+	// TextMarshaler (the 'MarshalText' method) allows serialization of the client
+	// state for subsequent restoration. The marshalled value can be passed to `New`
+	// to restore Client state.
+	encoding.TextMarshaler
+
+	// Get returns a the randomness at `round` or an error over a result channel.
+	// One result will be returned, with exactly one of Err or Data populated.
+	func Get(ctx context.Context, round uint64) <-chan Result
+
+	// RoundAt will return the most recent round of randomness that will be available
+	// at time for the current client.
+	func RoundAt(time time.Time) uint64
+}
+
+// Result represents the randomness for a single drand round.
+type Result struct {
+	Err error
+	Round uint64
+	Data []byte
+}
```

Comment:
I think we want to retain the `BeaconEntry` tuple as given by Lotus. That will be the core data type threaded throughout the calling code, after any error is known to be nil. You could embed a BeaconEntry in a Result to obtain the maybe-type semantics for a channel result (but see above re. channels).

Reply:
As a generic library, Drand isn't going to use a specific consumer implementation type. Moving the byte slice over to a beaconentry tuple seems reasonable enough?

Reply:
Right, yes I didn't mean depend on Lotus, but replicate the shape (after which Lotus can replace theirs with this one if they want).